lexer.cpp

```diff
@@ -124,6 +124,7 @@ Token Lexer::next()
         case ',':
         case ';':
         case '=':
+        case '#':
```

Comment:
Do you think we should support `#  ifdef` or error out?
If we need to error out, then we just need to add a check that the next character following '#' is alnum.

Reply:
I think it makes sense to error out and reduces the complexity of the grammar. I'll do that, thanks!

Reply:
Just did quick experiments, both gcc and clang allow spaces between the keyword and #. I'll leave this as it is for now.

Reply:
Oh. Interesting. I didn't know that. Then I think the current one is the right approach.
Referring to GCC [doc](https://gcc.gnu.org/onlinedocs/gcc-2.95.3/cpp_1.html#SEC3):
> Whitespace is also allowed before and after the `#'.